### PR TITLE
tests: add assert test application

### PR DESCRIPTION
**Pull request by kaspar030**
see #2931.

### CODE REVIEW

**Comment by authmillenon** [comment]
Do we really need a test application for this?

**Comment by OlegHahm** [comment]
I would say it makes sense to check if the feature works on all platforms. However, I would vote for making this a unittest. To be honest, most test applications (in `tests` folder) are still pretty pointless apart from acting as prove-of-concept.

**Comment by authmillenon** [comment]
How can we make this a unittest, if

1. a failing assert might halt the program (depending on the platform)
2. the actual action of `assert()` depends on a flag given at compile time

especially the second case would require 2 applications, one where the flag is set and one where it is not.

File: tests/assert/main.c [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ */
+
+#include <stdio.h>
+#include <assert.h>
+
+int main(void)
+{
+    printf("assert test app\n");
+
+    int i = 1;
```
Comment (Kijewski) [inline tests/assert/main.c]: `+volatile` to disable compiler optimizations.

**Comment by OlegHahm** [comment]
@kaspar030, please see and obey point 10 from the [Development Procedures](https://github.com/RIOT-OS/RIOT/wiki/Development-procedures)

**Comment by kaspar030** [comment]
Closed as not needed.

**Pull request closed without merging**